clients/python/tests/regression_test.py

```diff
@@ -44,3 +45,29 @@ def test_get_model_without_user_token(setup_env_user_token, client):
     assert rm.id
     assert (_rm := client.get_registered_model(name))
     assert rm.id == _rm.id
+
+
+@pytest.mark.e2e
+def test_get_few_registered_models(client: ModelRegistry):
+    """Test regression for paging without next page token.
+
+    Reported on: https://github.com/kubeflow/model-registry/issues/348
+    """
+    models = 9
+
+    for name in [f"test_model{i}" for i in range(models)]:
+        client.register_model(
+            name,
+            "s3",
+            model_format_name="test_format",
+            model_format_version="test_version",
+            version="1.0.0",
+        )
+
+    i = 0
+    for rm in client.get_registered_models().limit(10):
+        print(f"found {rm}")
+        i += 1
+        assert i < models + 1
+
+    assert i == models
```

Comment:
can you kindly consider making (possibly same test) a loop with plain:  

```py
for rm in client.get_registered_models():
```

and no limit, this way we ensure to cover codepath as well for unset limit. wdyt?